src/ffs/normalizer/versions/v1_0_0.ts

```diff
@@ -0,0 +1,73 @@
+import { CID, FileContent } from '../../../ipfs'
+import { Metadata, Header, TreeData, PrivateTreeData, PinMap } from '../../types'
+import check from '../../types/check'
+import basic from '../basic'
+import header from '../header'
+import { isNum, isBool } from '../../../common'
+import { defaultError } from '../errors'
+
+export const getFile = async (cid: CID, key?: string): Promise<FileContent> => {
+  const indexCID = await basic.getLinkCID(cid, 'index', key)
+  if(!indexCID) {
+    throw new Error("File does not exist")
+  }
+  return basic.getFile(indexCID, key)
+}
+
+export const getTreeData = async (cid: CID, key?: string): Promise<TreeData | PrivateTreeData> => {
+  const indexCID = await basic.getLinkCID(cid, 'index', key)
+  if(!indexCID) {
+    throw new Error(`Links do not exist: ${indexCID}`)
+  }
+  const links = await basic.getLinks(indexCID, key)
+  const childKey = key ? await header.getChildKey(cid, key) : undefined
```

Comment:
Just double checking: `undefined`, not `null`?

Reply:
Yup. A lot of these methods do essentially the same thing but with/without encryption so they take an optional `key` param
For instance:
```ts
export const getMetadata = async (cid: CID, key?: string): Promise<Metadata>
```

So here I'm getting the childKey that would be passed to `getMetadata`. Since it's an optional param, it is type `string | undefined` instead of `null`

This could be cleaner with the `Maybe` type that we've been talking about adding. Would force more intentionality :thinking: